### PR TITLE
Report better logs for github issue

### PR DESCRIPTION
**Pull request by yue9944882**
@sttts 

Plus some minor refactor 😃

Showing log in a better way for the bot.

### CODE REVIEW

File: cmd/publishing-bot/github.go [hunk 1]
```diff
@@ -53,7 +53,20 @@ func ReportOnIssue(e error, logs, token, org, repo string, issue int) error {
 	}
 
 	// create new newComment
-	body := fmt.Sprintf("The last publishing run failed: %v\n\n```%s```\n", e, tail(logs, 65000))
+	// body := fmt.Sprintf("The last publishing run failed: %v\n\n```%s```\n", e, tail(logs, 65000))
```
Comment (sttts) [inline cmd/publishing-bot/github.go]: no need for the old code, just drop the line

File: cmd/publishing-bot/log_builder.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+			logBuilder.logs = append(logBuilder.logs, rawLogs[i])
+			break
+		}
+		logBuilder.logs = append(logBuilder.logs, rawLogs[i])
+		size += len(rawLogs)
+	}
+	return logBuilder
+}
+
+func (builder *logBuilder) addHeading(line string) *logBuilder {
```
Comment (sttts) [inline cmd/publishing-bot/log_builder.go]: would make the methods which are supposed to be used upper-case

File: cmd/publishing-bot/github.go [hunk 3]
```diff
@@ -53,7 +53,20 @@ func ReportOnIssue(e error, logs, token, org, repo string, issue int) error {
...
+
+	body := newLogBuilderWithMaxBytes(65000, logs).
+		addHeading(fmt.Sprintf("The last publishing run failed: %v", e)).
+		addHeading("```").
+		trim("\n").
+		split("\n").
+		filter(func(line string) bool {
```
Comment (sttts) [inline cmd/publishing-bot/github.go]: does this mean that all non-`+` lines are skipped?

Comment (yue9944882) [reply cmd/publishing-bot/github.go]: @sttts yep. isn't that the behavior we want?

Comment (yue9944882) [reply cmd/publishing-bot/github.go]: the skipped lines could be found in the file logs

Comment (sttts) [reply cmd/publishing-bot/github.go]: Instead we want to show up to 50 `+` lines from the bottom, including all non-`+` lines. Above that we only show 50 non-`+` lines.

Comment (yue9944882) [reply cmd/publishing-bot/github.go]: got it

File: cmd/publishing-bot/github.go [hunk 4]
```diff
@@ -109,49 +110,26 @@ func CloseIssue(token, org, repo string, issue int) error {
...
-				return "..." + lines[0][max(0, len(lines[0])-maxBytes+3):]
+func transfromLogToGithubFormat(original string, maxLines int, headings ...string) string {
+	logCount := 0
+	transformed := NewLogBuilderWithMaxBytes(65000, original).
+		AddHeading(headings...).
+		AddHeading("```").
+		Trim("\n").
+		Split("\n").
+		Reverse().
+		Filter(func(line string) bool {
```
Comment (sttts) [inline cmd/publishing-bot/github.go]: is this working backwards, i.e. we get the last lines?

Comment (yue9944882) [reply cmd/publishing-bot/github.go]: yeah. it's a little tricky here. i did a reverse before filtering and reverse again to restore the order.

Comment (yue9944882) [reply cmd/publishing-bot/github.go]: plz see the test i wrote for it

**Pull request merged**